ambry-tools/src/main/java/com.github.ambry/clustermap/HelixBootstrapUpgradeTool.java

```diff
@@ -0,0 +1,466 @@
+          ofType(String.class);
+
+      ArgumentAcceptingOptionSpec<String> localDcOpt = parser.accepts("localDc", "The local datacenter name")
+          .withRequiredArg()
+          .describedAs("local_dc")
+          .ofType(String.class);
+
+      OptionSet options = parser.parse(args);
+      String hardwareLayoutPath = options.valueOf(hardwareLayoutPathOpt);
+      String partitionLayoutPath = options.valueOf(partitionLayoutPathOpt);
+      String zkLayoutPath = options.valueOf(zkLayoutPathOpt);
+      ArrayList<OptionSpec> listOpt = new ArrayList<>();
+      listOpt.add(hardwareLayoutPathOpt);
+      listOpt.add(partitionLayoutPathOpt);
+      listOpt.add(zkLayoutPathOpt);
+      ToolUtils.ensureOrExit(listOpt, options, parser);
+      bootstrapOrUpgrade(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath, options.valueOf(localDcOpt));
+    } catch (Exception e) {
+      System.out.println("Exception while executing partition command: " + e);
+    }
+  }
+
+  /**
+   * Takes in the path to the files that make up the static cluster map and adds or updates the cluster map information
+   * in Helix to make the two consistent.
+   * @param hardwareLayoutPath the path to the hardware layout file.
+   * @param partitionLayoutPath the path to the partition layout file.
+   * @param zkLayoutPath the path to the zookeeper layout file.
+   * @param localDc the name of the local datacenter. This can be null.
+   * @throws IOException if there is an error reading a file.
+   * @throws JSONException if there is an error parsing the JSON content in any of the files.
+   */
+  static void bootstrapOrUpgrade(String hardwareLayoutPath, String partitionLayoutPath, String zkLayoutPath,
+      String localDc) throws IOException, JSONException {
+    HelixBootstrapUpgradeTool clusterMapToHelixMapper =
+        new HelixBootstrapUpgradeTool(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath, localDc);
+    clusterMapToHelixMapper.updateHelix();
+  }
+
+  /**
+   * Instantiates this class with the given information.
+   * @param hardwareLayoutPath the path to the hardware layout file.
+   * @param partitionLayoutPath the path to the partition layout file.
+   * @param zkLayoutPath the path to the zookeeper layout file.
+   * @param localDc the name of the local datacenter. This can be null.
+   * @throws IOException if there is an error reading a file.
+   * @throws JSONException if there is an error parsing the JSON content in any of the files.
+   */
+  private HelixBootstrapUpgradeTool(String hardwareLayoutPath, String partitionLayoutPath, String zkLayoutPath,
+      String localDc) throws IOException, JSONException {
+    this.localDc = localDc;
+    parseZkJsonAndPopulateZkInfo(zkLayoutPath);
+
+    String partitionLayoutString = null;
+    try {
+      partitionLayoutString = Utils.readStringFromFile(partitionLayoutPath);
+    } catch (FileNotFoundException e) {
+      System.out.println("Partition layout path not found. Creating new file");
```

Comment:
could we instead do
```
String partitionLayoutString = null;
if (new File(partitionLayoutPath).exists()) {
  partitionLayoutString = Utils.readStringFromFile(partitionLayoutPath);
}
```
?

I would vote against using exceptions for control flow.